### PR TITLE
APP-4546 Bug fixes

### PR DESCRIPTION
**Pull request by symphony-annaLlorens**
### Fixes:

- Loading button fixed size
- Right icon well displayed
- Added button role

### CODE REVIEW

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -46,18 +46,32 @@ export const Button: React.FC<ButtonProps> = ({
     `${loading ? 'loading' : ''}`,
     `${prefix}--${size}`
   );
-  if(variant==='destructive') {
+
+  const [width, setWidth] = React.useState(0);
+  const ref = React.useRef(null);
+  React.useEffect(() => {
+    if (ref.current && ref.current.getBoundingClientRect().width) {
```
Comment (symphony-adnane) [inline src/components/button/Button.tsx:53]: can't we do this with CSS only? 
What are we trying to achieve?

Comment (symphony-annaLlorens) [reply src/components/button/Button.tsx:53]: To get dinamically the width of the button, this way when the button is loading it keep the same size

Comment (symphony-annaLlorens) [reply src/components/button/Button.tsx:53]: To get dynamically the width of the button, this way when the button is loading it keep the same size

Comment (AliciaSymphony) [reply src/components/button/Button.tsx:53]: I think that you can put the text to hidden, and center the loading icon with absolute

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -46,18 +46,32 @@ export const Button: React.FC<ButtonProps> = ({
...
+      role="button"
       className={classes}
       disabled={loading || disabled}
       /* eslint-disable react/button-has-type */
       type={type}
+      style={loading && children ? {width: `${width}px`}: {}}
       {...rest}
     >
-      {loading ? <i className="animate-spin tk-icon-loading" /> : iconLeft ? <>{iconLeft}{children}</> : iconRight ? <>{iconRight}{children}</> : children}
+      {loading ? <i className="animate-spin tk-icon-loading" /> : iconLeft ? <>{iconLeft}{children}</> : iconRight ? <>{children}{iconRight}</> : children}
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:74]: You can simply put

```
{loading ? <i className="animate-spin tk-icon-loading" /> :
  <i className="animate-spin tk-icon-loading" /> :  <>{iconLeft ? iconLeft : null}{children}{iconRight ? iconRight : null}</>
}
```

I don't think that we should restrict to left or right, if the user want to put both?

File: src/components/button/Button.tsx (hunk 1 above)
Comment (symphony-annaLlorens) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx (hunk 2 above)
Comment (symphony-annaLlorens) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx [hunk 3]
```diff
@@ -41,23 +41,28 @@ export const Button: React.FC<ButtonProps> = ({
...
   );
-  if(variant==='destructive') {
+
+  if(variant === 'destructive') {
     console.warn('The button variant: \'destructive\' will be deprecated.\n Please use: \'primary-destructive\' instead')
   }
+
   return (
     <button
+      role="button"
```
Comment (symphony-jean-michael) [inline src/components/button/Button.tsx:54]: Just curious. Why do you need to add the role 'button' on a button tag? (See https://stackoverflow.com/questions/26934719/do-i-need-role-button-on-a-button)

File: src/components/button/Button.tsx [hunk 4]
```diff
@@ -41,23 +41,28 @@ export const Button: React.FC<ButtonProps> = ({
...
       className={classes}
       disabled={loading || disabled}
       /* eslint-disable react/button-has-type */
       type={type}
       {...rest}
     >
-      {loading ? <i className="animate-spin tk-icon-loading" /> : iconLeft ? <>{iconLeft}{children}</> : iconRight ? <>{iconRight}{children}</> : children}
+      {loading && <i className="animate-spin tk-icon-loading" />}
+      <span className={classNames(iconRight && `${prefix}--icon-right`,iconLeft && `${prefix}--icon-left`)}
```
Comment (symphony-jean-michael) [inline src/components/button/Button.tsx:62]: IMO, classNames should take an object in parameter, isn't it ?
```classNames({ iconRight: `${prefix}--icon-right`, iconLeft: `${prefix}--icon-left` })```
(Se: https://www.npmjs.com/package/classnames)

**Review by AliciaSymphony — APPROVED** [review]

**Review by symphony-jean-michael — APPROVED** [review]

**Pull request merged**